Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -246,6 +247,34 @@ public struct TargetSourcesBuilder {
         }
     }
 
+    private func diagnoseMissingResources(in resources: [Resource]) {
+        // Stop early if the valid resources array count is the same as the user defined list
+        if resources.count == self.target.resources.count {
+            return
+        }
+        
+        // Filter the package list of resources down to those which are not included in the valid resources array
+        let invalidResources = self.target.resources.filter { targetResource in
+            return resources.contains(where: { resource in
+                resource.path.relative(to: self.targetPath).pathString == targetResource.path
```

Comment:
Would this fail for things like `./File` in the manifest, since the normalized path provided by AbsolutePath wouldn't match the literal string on the right-hand side (the `path` in `targetResource`)?  I wonder if both should be normalized to relative paths.  I would have to look into this a bit more to suggest something more concrete, but I'm wondering if there might be an issue here.

Reply:
Good spot! Will add a test for this case and see what I can do to address this.

Reply:
I've added a test for this in the latest commit and it's all passing with no further changes to the code - it may not be immediately clear but it does seem that it's handling it. Let me know what you think